tests/test_actions.py

```diff
@@ -996,6 +996,10 @@ def test_log_new_events_prints_correct_event(self, mock_describe_events):
                     "LogicalResourceId": "id-2",
```

Comment:
This test doesn't actually test that the new log events are logged. Right now, it's not really testing anything at all. You should update this test to actually test for the behavior that we expect log_new_events to have.

Reply:
Hey @jfalkenstein 
Thank you again for your time and effort.

Yeah, that was the first try run. I know I need to update the integration test code to accomodate my new 'log_new_events'. Was thinking there might be 1 or 2 failures and then I could fix them without too much efforts. Unfortunately, I saw almost half of the testing scenarios failed and was not quite sure where to start to update the integration tests code. Could you please shed some lights on this?

Thank you.
Feng

Reply:
Yeah, you'll want to set your region to eu-west-1 when running integration tests. Otherwise, a number of them will fail.

Reply:
Hey @jfalkenstein 
Thank you for your response.

To try run the integration tests, I enter below command in the project root directory...

```
[fli@almalinux sceptre]$ pwd
/home/fli/github/sceptre
[fli@almalinux sceptre]$ 

[fli@almalinux sceptre]$ AWS_PROFILE=sceptre AWS_DEFAULT_REGION=eu-west-1 poetry run behave --junit --junit-directory build/behave integration-tests
...
[fli@almalinux sceptre]$ 
```

The integration test reports were collected successfully ...
```
[fli@almalinux sceptre]$ ls build/behave/
TESTS-features.create-change-set.xml               TESTS-sceptre.integration-tests.features.create-change-set.xml
TESTS-features.create-stack.xml                    TESTS-sceptre.integration-tests.features.create-stack.xml
TESTS-features.delete-change-set.xml               TESTS-sceptre.integration-tests.features.delete-change-set.xml
TESTS-features.delete-stack-group.xml              TESTS-sceptre.integration-tests.features.delete-stack-group.xml
TESTS-features.delete-stack.xml                    TESTS-sceptre.integration-tests.features.delete-stack.xml
TESTS-features.dependency-resolution.xml           TESTS-sceptre.integration-tests.features.dependency-resolution.xml
TESTS-features.describe-change-set.xml             TESTS-sceptre.integration-tests.features.describe-change-set.xml
TESTS-features.describe-stack-group-resources.xml  TESTS-sceptre.integration-tests.features.describe-stack-group-resources.xml
TESTS-features.describe-stack-group.xml            TESTS-sceptre.integration-tests.features.describe-stack-group.xml
TESTS-features.describe-stack-resources.xml        TESTS-sceptre.integration-tests.features.describe-stack-resources.xml
TESTS-features.drift.xml                           TESTS-sceptre.integration-tests.features.drift.xml
TESTS-features.dump-template-s3.xml                TESTS-sceptre.integration-tests.features.dump-template-s3.xml
TESTS-features.dump-template.xml                   TESTS-sceptre.integration-tests.features.dump-template.xml
TESTS-features.execute-change-set.xml              TESTS-sceptre.integration-tests.features.execute-change-set.xml
TESTS-features.launch-stack-group.xml              TESTS-sceptre.integration-tests.features.launch-stack-group.xml
TESTS-features.launch-stack.xml                    TESTS-sceptre.integration-tests.features.launch-stack.xml
TESTS-features.list-change-sets.xml                TESTS-sceptre.integration-tests.features.list-change-sets.xml
TESTS-features.lock-stack.xml                      TESTS-sceptre.integration-tests.features.lock-stack.xml
TESTS-features.project-dependencies.xml            TESTS-sceptre.integration-tests.features.project-dependencies.xml
TESTS-features.prune.xml                           TESTS-sceptre.integration-tests.features.prune.xml
TESTS-features.stack-diff.xml                      TESTS-sceptre.integration-tests.features.stack-diff.xml
TESTS-features.stack-output-external-resolver.xml  TESTS-sceptre.integration-tests.features.stack-output-external-resolver.xml
TESTS-features.stack-output-resolver.xml           TESTS-sceptre.integration-tests.features.stack-output-resolver.xml
TESTS-features.unlock-stack.xml                    TESTS-sceptre.integration-tests.features.unlock-stack.xml
TESTS-features.update-stack.xml                    TESTS-sceptre.integration-tests.features.update-stack.xml
TESTS-features.validate-template.xml               TESTS-sceptre.integration-tests.features.validate-template.xml
[fli@almalinux sceptre]$ 
```

A typical passed report:

[TESTS-features.create-change-set.xml.log](https://github.com/Sceptre/sceptre/files/12297310/TESTS-features.create-change-set.xml.log)

A typical failed report:

[TESTS-features.create-stack.xml.log](https://github.com/Sceptre/sceptre/files/12297312/TESTS-features.create-stack.xml.log)

I reckon I need to update the integration code somewhere, but not very sure. Could you please advise ?

Thank you.
Feng

Reply:
According to those failure logs you linked to... it looks like HookType isn't always returned in the cloudformation events. I don't think you need to fix the integration test code. I think you need to fix the event code to handle cases where there's no HookType key in the results.

Reply:
Hey @jfalkenstein 

Thank you for your help. I am editing my change as below:

```
PS C:\Users\fli\github\sceptre> git diff .\sceptre\plan\actions.py
diff --git a/sceptre/plan/actions.py b/sceptre/plan/actions.py
index bebd09c..78d599d 100644
--- a/sceptre/plan/actions.py
+++ b/sceptre/plan/actions.py
@@ -840,21 +840,34 @@ class StackActions:
         events.reverse()
         new_events = [event for event in events if event["Timestamp"] > after_datetime]
         for event in new_events:
-            self.logger.info(
-                " ".join(
-                    [
-                        self.stack.name,
-                        event["LogicalResourceId"],
-                        event["ResourceType"],
-                        event["ResourceStatus"],
-                        event.get("ResourceStatusReason", ""),
-                        event["HookType"],
-                        event["HookStatus"],
-                        event.get("HookStatusReason", ""),
-                        event["HookFailureMode"],
-                    ]
+            if "HookStatus" in event:
+                self.logger.info(
+                    " ".join(
+                        [
+                            self.stack.name,
+                            event["LogicalResourceId"],
+                            event["ResourceType"],
+                            event["ResourceStatus"],
+                            event.get("ResourceStatusReason", ""),
+                            event["HookType"],
+                            event["HookStatus"],
+                            event.get("HookStatusReason", ""),
+                            event["HookFailureMode"],
+                        ]
+                    )
+                )
+            else:
+                self.logger.info(
+                    " ".join(
+                        [
+                            self.stack.name,
+                            event["LogicalResourceId"],
+                            event["ResourceType"],
+                            event["ResourceStatus"],
+                            event.get("ResourceStatusReason", ""),
+                        ]
+                    )
                 )
-            )
             after_datetime = event["Timestamp"]
         return after_datetime

PS C:\Users\fli\github\sceptre>
```

Will see how it is going. Cheers!

Feng